auth-server/src/test/java/com/makeurpicks/service/OAuthServiceTest.java

```diff
@@ -0,0 +1,210 @@
+import com.makeurpicks.domain.OAuthClientDetailsBuilder;
+import com.makeurpicks.exception.OAuthclientValidationException;
+import com.makeurpicks.exception.OAuthclientValidationException.OAuthClientExceptions;
+
+//@RunWith(SpringJUnit4ClassRunner.class)
+//@SpringApplicationConfiguration(classes = AuthServerApplication.class)
+@RunWith(MockitoJUnitRunner.class)
+
+
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class OAuthServiceTest {
+
+	
+	@Autowired
+	@InjectMocks
+	private OAuthClientsService oAuthClientService;
+	
+	
+	
+	
+	//@InjectMocks
+	//@DataJpaTest
+	@Mock
+	public ClientDetailsDAO clientDetailsDAO;
+	
+
+	@Rule
+	public ExpectedException expectedEx = ExpectedException.none();
+	
+	@Before
+	public void setup() {
+		MockitoAnnotations.initMocks(this);
+		
+		/*OAuthClientDetails authClientDetails=new OAuthClientDetails();
+		authClientDetails.setClientId("1");
+		authClientDetails.setClient_secret("111111");
+		
+		OAuthClientDetails authClientDetails2=new OAuthClientDetails();
+		authClientDetails.setClientId("2");
+		authClientDetails.setClient_secret("22222");
+		clientDetailsDAO.save(authClientDetails);
+		clientDetailsDAO.save(authClientDetails2);*/
+	}
+	
+	@Test
+	public void testGetAllClients(){
+		if(oAuthClientService==null){
+			fail();
+		}
+		OAuthClientDetails oAuthClientDetails1=new OAuthClientDetailsBuilder("123123", "testsecret", "ROLE_TRUSTED_CLIENT","client_credentials,password,authorization_code,refresh_token", "read,write","http://localhost", Boolean.FALSE.toString()).build();
+		OAuthClientDetails oAuthClientDetails2=new OAuthClientDetailsBuilder("123123", "testsecret", "ROLE_TRUSTED_CLIENT","client_credentials,password,authorization_code,refresh_token", "read,write","http://localhost", Boolean.FALSE.toString()).build();
+		List<OAuthClientDetails> stubReturnList=new ArrayList<OAuthClientDetails>();
+		
+		stubReturnList.add(oAuthClientDetails1);
+		stubReturnList.add(oAuthClientDetails2);
+		when(clientDetailsDAO.findAll()).thenReturn(stubReturnList);
+		Iterable<OAuthClientDetails > clients=oAuthClientService.findAllClients();
+		List<OAuthClientDetails> oauthclientdetailslist=new ArrayList<OAuthClientDetails>();
+		if(clients!=null){
+			clients.forEach(oAuthClientDetails -> oauthclientdetailslist.add(oAuthClientDetails));	
+		}
+		
+		//assertNotEquals(Long.valueOf(0).longValue(), Long.valueOf(oauthclientdetailslist.size()).longValue());
+		//if()
+		assertThat(oauthclientdetailslist.size()==2);
+		
+		
+	}
+	
+	
+	@Test
+	public void testCreateClientsInValidDataNoClientId() throws OAuthclientValidationException {
+		if(oAuthClientService==null){
+			fail();
+		}
+		expectedEx.expect(OAuthclientValidationException.class);
+		expectedEx.expectMessage(OAuthClientExceptions.CLIENT_ID_NULL.toString());
+		//assertThat(expectedEx.e)
+		//SINCE THERE COULD BE MANY REASONS WITHINA SINGLE EXCEPTION BLOCK, GET MESSAGES RETURN A ARRAY. COMPARING IT WITH A 
```

Comment:
Hi Pawal,
I agree with your comments. The validation method etc were created for a different task (save ClientDetails to Database from in-memory) and I just got the code and used it (I did not get it from master branch or development branch as that was also under development along with my ticket). Though I do not mind updating it, I just wanted to update you about background of the method and highlight that it is not entirely a design mistake from my end. I did not change it explicitly because when you do pull requests from different branches, there could be conflict with one branch having the method as private and other as public.

Thanks
Hari